### PR TITLE
refactor: update RimeWithWeasel.cpp, style holded by session

### LINKED ISSUE #1066: `color_scheme_dark` 配置不能立即生效
**上报前请检查**
- [x] 我遇到的问题没有其他人在 issue 里提到过
- [x] 我的小狼毫版本于 rime/weasel 下载
- [x] 我在使用小狼毫的最新发布版本，或最新发布版本后的 CI 构建

**操作系统信息**
 - OS 详细版本: [ Windows 11 23H2 22631.2861]
 - 小狼毫版本: [ 最新的 CI 版本（[1e78320](https://github.com/rime/weasel/releases/tag/latest)） ]

**描述遇到的问题**
`color_scheme_dark` 配置不能立即生效

**复现步骤**
重现问题的步骤，如：
1. 现在在 Windows 设置里面是「浅色」，输入法是 `color_scheme` 的配色
2. 改变设置为「深色」，输入法依旧是 `color_scheme` 的配色，必须手动「重启算法服务」后，输入法才能变成 `color_scheme_dark` 的配色
3. 从「深色」切换成「浅色」，也需要手动「重启算法服务」

**预期行为**
Windows 颜色改变后，应该立即切换成配置文件对应的配色

**用户文件**
`weasel.custom.yaml` 的配置如下：
```
patch:
  "style/color_scheme": google
  "style/color_scheme_dark": lost_temple
```

### PR DESCRIPTION
**Pull request by fxliang**
- fix: style incorrect in some case when dark mode switches. introduced by #1046, fix #1066 
- refactor: show_notifications_when #1031, keyword is changed to `show_notifications`, bool type or list of { option_name in schema / keyword 'schema'}. it can be set up in both weasel.yaml(weasel.custom.yaml) and schema file.
- feat: notifications tips for customized switches
- feat: global ascii mode, with param `global_ascii: {bool}`
- feat: show_notifications_time customizable with param `show_notifications_time: {int}`, unit: ms
- chore: UI function de-warning.

```yaml
# configuration: when to show notifications
show_notifications:  true    # default true to enable all tips, false to disable all tips (except deployment's)
# list type of setting
#show_notifications:
#  - ascii_mode 
#  - ascii_punct
#  - full_shape
#  - simplification
#  - customized_option_name_in_schema
#  - schema    # this is for tip after schema switched

# how long a period notifications shows
show_notifications_time: 1200
# use global ascii status, set true means all session in the same ascii mode all the time.
global_ascii: false
```

global_ascii
all_tips
dark mode scheme auto switching, with schema specified color scheme and dark color scheme.
dark_mode_auto_switching

### CODE REVIEW

File: output/data/weasel.yaml [hunk 1]
```diff
@@ -10,9 +10,13 @@ app_options:
     ascii_mode: true
 
 # configuration: when to show notifications
-# always, never, or combination of: ascii_mode, ascii_punct, shape, simplification, schema
-# e.g., ascii_mode+shape+schema
+# always, never, or combination of: ascii_mode, ascii_punct, full_shape, simplification, schema
+# e.g., ascii_mode+full_shape+schema
 show_notifications_when: always
```
Comment (lotem) [inline output/data/weasel.yaml:15]: 鼠鬚管的 `show_notifications_when` 不是這個意思吧？

File: RimeWithWeasel/RimeWithWeasel.cpp [hunk 2]
```diff
@@ -956,34 +990,35 @@ static void _RimeGetStringWithFunc(RimeConfig* config, const char* key, std::wst
 		value = *fallback;
 }
 
-void _UpdateShowNotificationsWhen(RimeConfig* config, UINT* show_notifications_when)
-{
-	char buffer[256] = { 0 };
+void RimeWithWeaselHandler::_UpdateShowNotificationsWhen( RimeConfig *config) {
+	char buffer[512] = {0};
```
Comment (lotem) [inline RimeWithWeasel/RimeWithWeasel.cpp:994]: 處理成 List 更好。明明配置有這個能力，何必再去做字符串匹配呢。

首先，`show_notifications_when` 不見得是做這個用的。得查查刪掉的代碼。

File: output/data/weasel.yaml (hunk 1 above)
Comment (fxliang) [reply output/data/weasel.yaml:15]: 不知道，又没有人解释，只好自己理解做个雏形来了。等解析（期待脸）

File: RimeWithWeasel/RimeWithWeasel.cpp (hunk 2 above)
Comment (fxliang) [reply RimeWithWeasel/RimeWithWeasel.cpp:994]: 我了解下List怎么用

**Pull request merged**